src/main/java/com/google/codeu/servlets/StatsPageServlet.java

```diff
@@ -1,41 +1,128 @@
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-
-import com.google.codeu.data.Datastore;
-import com.google.gson.JsonObject;
-
-/**
- * Handles fetching site statistics.
- */
-@WebServlet("/stats")
-public class StatsPageServlet extends HttpServlet{
-
-  private Datastore datastore;
-
-  @Override
-  public void init() {
-    datastore = new Datastore();
-  }
-
-  /**
-   * Responds with site statistics in JSON.
-   */
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {
-
-    response.setContentType("application/json");
-
-    int messageCount = datastore.getTotalMessageCount();
-
-    JsonObject jsonObject = new JsonObject();
-    jsonObject.addProperty("messageCount", messageCount);
-    response.getOutputStream().println(jsonObject.toString());
-  }
-}
\ No newline at end of file
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.AbstractCollection;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.mail.MailService.Message;
+import com.google.codeu.data.Datastore;
+import com.google.gson.JsonObject;
+
+/**
+ * Handles fetching site statistics.
+ */
+@WebServlet("/stats")
+public class StatsPageServlet extends HttpServlet{
+
+  private Datastore datastore;
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Responds with site statistics in JSON.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("application/json");
+    
+    com.google.codeu.data.Message message = new com.google.codeu.data.Message("Alan", "Ola k ase");
+    com.google.codeu.data.Message message2 = new com.google.codeu.data.Message("Pepe", "Esta chido el CodeU");
+    com.google.codeu.data.Message message3 = new com.google.codeu.data.Message("Pepe", "Que me voy al GooglePlex!!!!!!");
+    
+    //List of all messages
+    List<com.google.codeu.data.Message> allmessages = datastore.getAllMessages();
+    allmessages.add(message);
+    allmessages.add(message2);
+    allmessages.add(message3);
+    
+    //Json for the JS
+    JsonObject jsonObject = new JsonObject();
+    
+    //Number of messages
+    int messageCount = datastore.getTotalMessageCount();
+    
+    //Number of messages property added to the Json
+    jsonObject.addProperty("messageCount", messageCount);
+    
+    //Store of the biggest message
+    com.google.codeu.data.Message biggest = allmessages.get(0);
+	
+    //Checking the size of the list
+	if(allmessages.size()==0) {
+		
+		jsonObject.addProperty("messageAvg", 0); 

```

Comment:
I'd suggest creating local variables for each of these computed values and then just move the code that assigns these local variables to properties at the bottom of this function. This has a few benefits over what you have here.

1. It makes sure all properties are assigned no matter the number of messages.
2. It allows you to use the Java type system to enforce what the properties should be. For example here you have the "biggestMessage" property set with an integer in one case and a string in another. If had a biggestMessage integer or String local variable, the compiler would have caught your inconsistent setting of this property.
3. It makes adding new properties less error prone because you only have to update the jsonObject in a single location at the bottom.

Reply:
@ArgaAlan I'm not completely sure that I follow these changes and suggestions; did you chose to implement this in your program? If so, could you possibly point me to where?

Reply:
> @ArgaAlan I'm not completely sure that I follow these changes and suggestions; did you chose to implement this in your program? If so, could you possibly point me to where?

I did, before I first checked if the list was empty but I finally change it to compute everything and just add it to the JSON

Reply:
oooh, so basically, you computed the variables, and then added them all to the json after?